### PR TITLE
Add predict_proba and threshold in cnn,bilstm

### PR DESCRIPTION
**Pull request by nsorros**
Description
---

Add proba function to NN to be able to return probabilities and also add threshold to be able to predict with different thresholds

Checklist
---

- [x] Added link to Github issue or Trello card
- [x] Added tests

### CODE REVIEW

**Comment by nsorros** [comment]
[Add proba function and threshold parameter to NN architectures to be able to lower the threshold for predictions](https://trello.com/c/ZnVdrg6B/1133-add-proba-function-and-threshold-parameter-to-nn-architectures-to-be-able-to-lower-the-threshold-for-predictions)

File: tests/test_bilstm.py [hunk 1]
```diff
@@ -162,3 +162,41 @@ def test_early_stopping():
...
+
+
+def test_predict_proba():
+    X = [
+        "One",
+        "One only",
+        "Two nothing else",
+        "Two and three"
+    ]
+    Y = np.array([0, 0, 1, 1])
```
Comment (aCampello) [inline tests/test_bilstm.py:174]: Can we add a test where Y is a sparse array ? I think the tests are missing that part of the code.

**Review by aCampello — APPROVED** [review]
No major comments, just a suggestion about adding a sparse array to the tests so the code coverage doesn't miss it.

File: tests/test_bilstm.py (hunk 1 above)
Comment (nsorros) [reply tests/test_bilstm.py:174]: good point

**Comment by nsorros** [comment]
@aCampello btw i altered the code a bit since sparse_y does not make sense for probs which are dense. let me know if this does not make sense

File: tests/test_bilstm.py (hunk 1 above)
Comment (nsorros) [reply tests/test_bilstm.py:174]: As it turns out there is a test for sparse.
https://github.com/wellcometrust/WellcomeML/blob/6b7503a043157edf5e40d22b9c2c312d255b60ae/tests/test_cnn.py#L96

Did you mean about proba? I removed sparse_y from proba as it did not make sense

**Comment by aCampello** [comment]
Makes sense, actually!

**Pull request merged**